control-plane/subcommand/partition-init/command.go

```diff
@@ -0,0 +1,208 @@
+		"Toggle for using HTTPS for all API calls to Consul.")
+
+	c.flags.DurationVar(&c.flagTimeout, "timeout", 10*time.Minute,
+		"How long we'll try to bootstrap ACLs for before timing out, e.g. 1ms, 2s, 3m")
+	c.flags.StringVar(&c.flagLogLevel, "log-level", "info",
+		"Log verbosity level. Supported values (in order of detail) are \"trace\", "+
+			"\"debug\", \"info\", \"warn\", and \"error\".")
+	c.flags.BoolVar(&c.flagLogJSON, "log-json", false,
+		"Enable or disable JSON output format for logging.")
+
+	c.k8s = &k8sflags.K8SFlags{}
+	flags.Merge(c.flags, c.k8s.Flags())
+	c.help = flags.Usage(help, c.flags)
+
+	// Default retry to 1s. This is exposed for setting in tests.
+	if c.retryDuration == 0 {
+		c.retryDuration = 1 * time.Second
+	}
+}
+
+func (c *Command) Synopsis() string { return synopsis }
+
+func (c *Command) Help() string {
+	c.once.Do(c.init)
+	return c.help
+}
+
+// Run bootstraps ACLs on Consul servers and writes the bootstrap token to a
+// Kubernetes secret.
+// Given various flags, it will also create policies and associated ACL tokens
+// and store the tokens as Kubernetes Secrets.
+// The function will retry its tasks indefinitely until they are complete.
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+	if err := c.flags.Parse(args); err != nil {
+		return 1
+	}
+	if len(c.flags.Args()) > 0 {
+		c.UI.Error("Should have no non-flag arguments.")
+		return 1
+	}
+
+	// Validate flags
+	if err := c.validateFlags(); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+	var cancel context.CancelFunc
+	c.cmdTimeout, cancel = context.WithTimeout(context.Background(), c.flagTimeout)
+	// The context will only ever be intentionally ended by the timeout.
+	defer cancel()
+
+	var err error
+	c.log, err = common.Logger(c.flagLogLevel, c.flagLogJSON)
+	if err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	serverAddresses := c.flagServerAddresses
+	// Check if the provided addresses contain a cloud-auto join string.
+	// If yes, call godiscover to discover addresses of the Consul servers.
+	if len(c.flagServerAddresses) == 1 && strings.Contains(c.flagServerAddresses[0], "provider=") {
+		var err error
+		serverAddresses, err = godiscover.ConsulServerAddresses(c.flagServerAddresses[0], c.providers, c.log)
+		if err != nil {
+			c.UI.Error(fmt.Sprintf("Unable to discover any Consul addresses from %q: %s", c.flagServerAddresses[0], err))
+			return 1
+		}
+	}
+
+	scheme := "http"
+	if c.flagUseHTTPS {
+		scheme = "https"
+	}
+
+	serverAddr := fmt.Sprintf("%s:%d", serverAddresses[0], c.flagServerPort)
+	consulClient, err := consul.NewClient(&api.Config{
+		Address: serverAddr,
+		Scheme:  scheme,
+		Token:   "", // TODO: Figure out if a token is required here and which one that is.
+		TLSConfig: api.TLSConfig{
+			Address: c.flagConsulTLSServerName,
+			CAFile:  c.flagConsulCACert,
+		},
+	})
+	if err != nil {
+		c.log.Error(fmt.Sprintf("Error creating Consul client for addr %q: %s", serverAddr, err))
```

Comment:
Looks like the pattern you're following is to use `c.UI` for fatal issues and `c.log` for runtime logs. By that pattern this should use `c.UI`